doc/api/http.md

```diff
@@ -48,26 +48,39 @@ list like the following:
 added: v0.3.4
 -->
 
-The HTTP Agent is used for pooling sockets used in HTTP client
-requests.
-
-The HTTP Agent also defaults client requests to using
-`Connection: keep-alive`. If no pending HTTP requests are waiting on a
-socket to become free the socket is closed. This means that Node.js's
-pool has the benefit of keep-alive when under load but still does not
-require developers to manually close the HTTP clients using
-KeepAlive.
-
-If you opt into using HTTP KeepAlive, you can create an Agent object
-with that flag set to `true`.  (See the [constructor options][].)
-Then, the Agent will keep unused sockets in a pool for later use.  They
-will be explicitly marked so as to not keep the Node.js process running.
-However, it is still a good idea to explicitly [`destroy()`][] KeepAlive
-agents when they are no longer in use, so that the Sockets will be shut
-down.
-
-Sockets are removed from the agent's pool when the socket emits either
-a `'close'` event or a special `'agentRemove'` event. This means that if
+An `Agent` is responsible for managing connection persistence
+and reuse for HTTP clients. It maintains a queue of pending requests
+for a given host and port, reusing a single socket connection for each
+until the queue is empty, at which time the socket is destroyed.
+When a socket connection is no longer writable, for example, if a server
+closes the connection, the socket is destroyed.
+
+Unless an `Agent` is expressly provided in the [`http.request()`]
+options, all HTTP client requests will use the default
+[`http.globalAgent`].
+
+In addition to managing connection persistence for queued HTTP
```

Comment:
I agree that it's confusing. But this is how I read the code here https://github.com/nodejs/node/blob/master/lib/_http_agent.js#L47-L81. If there are pending requests they are reused and the `keepAlive` option is ignored. But if no requests are pending and `keepAlive` is truthy, when a socket emits a 'free' event, they are pooled. Otherwise, they are destroyed on line80. We can see the client emits 'free' here https://github.com/nodejs/node/blob/master/lib/_http_client.js#L551-L568 if `req.shouldKeepAlive` is false, and the request's `shouldKeepAlive` mirrors `agent.shouldKeepAlive` here https://github.com/nodejs/node/blob/master/lib/_http_client.js#L216-L222.

Confusing, but see also @bnoordhuis comment here https://github.com/nodejs/node/issues/10774#issuecomment-272470317